### ISSUE TITLE
Online Course Engagement

### ISSUE THREAD

**Issue by siddhant4ds** [abhisheks008/ML-Crate#696]
### ML-Crate Repository (Proposing new issue)
:red_circle: **Project Title** : Online Course Engagement
:red_circle: **Aim** : Predict course-completion status based on user engagement metrics on online course platform.
:red_circle: **Dataset** : [Predict Online Course Engagement Dataset](https://www.kaggle.com/datasets/rabieelkharoua/predict-online-course-engagement-dataset)
:red_circle: **Approach** : Try to use 3-4 algorithms to implement the models and compare all the algorithms to find out the best fitted algorithm for the model by checking the accuracy scores. Also do not forget to do a exploratory data analysis before creating any model.

**********************************************************************************
### 📍 **Follow the Guidelines to Contribute in the Project :**
* You need to create a separate folder named as the Project Title.
* Inside that folder, there will be four main components.
   * Images - To store the required images.
   * Dataset - To store the dataset or, information/source about the dataset.
   * Model - To store the machine learning model you've created using the dataset.
   * `requirements.txt` - This file will contain the required packages/libraries to run the project in other machines.
* Inside the `Model` folder, the `README.md` file must be filled up properly, with proper visualizations and conclusions.

***********************************************************************
:red_circle::yellow_circle: **Points to Note :**

- The issues will be assigned on a first come first serve basis, 1 Issue == 1 PR.
- "Issue Title" and "PR Title should be the same. Include issue number along with it.
- Follow Contributing Guidelines & Code of Conduct before start Contributing.

***********************************************************************
:white_check_mark: **To be Mentioned while taking the issue :**
- Full name : 
- GitHub Profile Link : 
- Participant ID (If not, then put NA) :
- Approach for this Project :
- What is your participant role? (Mention the Open Source Program name. Eg. HRSoC, GSSoC, GSOC etc.)

*************************************************************
Happy Contributing 🚀 

All the best. Enjoy your open source journey ahead. 😎

**Comment by siddhant4ds** [comment]
@abhisheks008 Please assign me this issue. Thanks.

Full name: Siddhant Tiwari
GitHub profile link: https://github.com/siddhant4ds
Participant ID: sid4ds (Devfolio), sid4ds (Discord)
Participant role: SSOC-3 Contributor
Approach:
1. Exploratory data analysis
2. Feature engineering using domain knowledge as well as automatic tools (e.g. OpenFE).
3. Implement simple models such as Logistic Regression, Naive Bayes, Decision Tree, etc for baseline performance
4. Implement gradient-boosting models (XGBoost, CatBoost,LightGBM) and neural networks (sklearn-MLP, Keras-Sequential)
5. Create Streamlit/Replit web app for inference using the best model.

**Comment by abhisheks008** [comment]
Assigned @siddhant4ds
